yoti/site/models/user.php

```diff
@@ -154,7 +154,8 @@ public function getUserIdByYotiId($yotiId)
             $query = $db->getQuery(TRUE)
                 ->select('joomla_userid')
                 ->from(YotiHelper::YOTI_USER_TABLE_NAME)
-                ->where('identifier=' . $db->quote($yotiId));
+                ->where('identifier=' . $db->quote($yotiId))
+                ->setLimit(1);
```

Comment:
If you're using `loadResult()` do you still need `->setLimit(1);`?

Reply:
I am setting the limit to optimise the query. So it's always a good practice to set limit to 1 if you are not expecting more than one result. Maybe loadAssoc will make more sense.

Reply:
I'm asking because it seems `loadResult` is meant for a single result: https://docs.joomla.org/Selecting_data_using_JDatabase#Single_Value_Result

Reply:
I think loadResult should be alright there, see ref link here 
https://docs.joomla.org/Selecting_data_using_JDatabase

This is often the result of a 'count' query to get a number of records:

or where you are just looking for a single field from a single row of the table (or possibly a single field from the first row returned).

Reply:
```
loadRow() returns an indexed array from a single record in the table:
loadAssoc() returns an associated array from a single record in the table:
```
`loadAssoc` definitely sounds better, but they both return a single row. That's why I'm wondering if that limit is necessary.

Reply:
I think this case loadResult is the right one to use. I am setting the limit to 1 so MySQL will stop searching from the first row it finds in the table that matches the query, otherwise it will continue searching the whole table even after finding the matching row which is slower and that’s not good. This is one way to optimise an SQL query.

Reply:
Yes, `loadResult` is the right one because you're only querying for the ID. What I'm saying is that I think it already limits the results to 1.